### ISSUE TITLE
Block inserter dialog has unexpected tab stop

### ISSUE THREAD

**Issue by getdave** [WordPress/gutenberg#47013]
As reported in https://github.com/WordPress/gutenberg/issues/46939#issuecomment-1374652632 the [Block inserter dialog has random tab stop](https://user-images.githubusercontent.com/444434/211516537-cd3d2838-0c1d-4ece-b5a4-3121e3e0ae6e.png). This needs to be removed.

See also https://github.com/WordPress/gutenberg/issues/24975

**Comment by ntsekouras** [comment]
Do you refer to the case when the container is scrollable? Because that it seems is how it needs to be implemented. See https://github.com/WordPress/gutenberg/pull/46580 and https://chromestatus.com/feature/5231964663578624.

**Comment by getdave** [comment]
> Do you refer to the case when the container is scrollable? Because that it seems is how it needs to be implemented. See #46580 and https://chromestatus.com/feature/5231964663578624.

Good question. I'm not sure. I'll take a snapshot of the point in the video from @alexstine post it here and see if it's related.

**Comment by scruffian** [comment]
This seems to be happening when the block inserter has scrollbars. The extra tab stop enables keyboard users to scroll the panel using the arrow keys. 

(Note: The block inserter only has scrollbars on Windows which seems like another issue we should fix)

@alexstine does it therefore make sense to keep this tab stop? I appreciate that it doesn't add value for blind users, but for sighted keyboard users it is serving a purpose.

**Comment by alexstine** [comment]
@scruffian No, it never makes sense to keep an empty tab stop with no meaningful label for screen readers. It is extremely confusing. Anyone up for doing some research and figuring out how to label it so it works for sighted users and screen reader users?

CC @joedolson 

Thanks.

**Comment by scruffian** [comment]
I'm happy to help solve this one, but I'm not sure what needs to done. I tried adding a label and a role to the place where the tab stop is here: https://github.com/WordPress/gutenberg/pull/47121. Feedback/direction welcome.

**Comment by joedolson** [comment]
An empty tab stop creates frustration - because a user has no idea what they're missing, and can't differentiate between "this is an accessibility error" and "this is not something I need." So even if the purpose of the tab stop is only for sighted keyboard users, it should be labeled for screen readers.

Also, since keyboard users may be sighted, the tab stop should have a visual cue to provide affordance - as it is, it theoretically provides access for sighted keyboard users, but there's really no way for those users to know that.

If this control is kept, I suggest two things:

1) The tab stop should become visible on focus. It could display arrows pointing up and down, for example. This gives visual affordance to what the tab stop does.
2) The control should be labeled with function: 'use arrow keys to scroll blocks up and down'. 

However, I'm not sure this control should be kept. What benefit does a keyboard user get from scrolling the list? If a user focuses the scrolling, then uses the down arrow to move to the bottom of the scrollable panel, then their next tab stop is still the first item in the list. So this feature is really only benefiting users who are mixed keyboard & pointer users - if they can scroll with the keyboard and then select with the mouse.

**Comment by jasmussen** [comment]
This tabstop seems like a bug, I don't think that inserter is meant to be scrollable, and in my interaction with it, it behaves expectedly as a modal:

* Focus set on the first tabbable
* Tab loops
* Tab to the list of blocks enables the roving arrowkey nav for selecting a blocok
* Esc closes and sets focus back on the opener:

GIF showing this:

![tab](https://user-images.githubusercontent.com/1204802/212297784-b3f0d508-fea6-407f-9642-ac0c617a46dc.gif)

Given there's a scrollbar visible in Windows, it seems like the bug to fix may be to simply make that scrollbar not appear (which is to say, find out what invokes it and fix that).

**Comment by joedolson** [comment]
@joen So, that's not the inserter I was testing; I was looking at this one:

![image](https://user-images.githubusercontent.com/1686451/212374397-8678f005-804f-4f99-b28b-3586573f420c.png)

Block/Pattern inserter with scrollbar.

**Comment by getdave** [comment]
Just noting the inserter @jasmussen is looking at is the one referenced in the Issue

See this screen shot 

https://user-images.githubusercontent.com/444434/211516537-cd3d2838-0c1d-4ece-b5a4-3121e3e0ae6e.png

**Comment by joedolson** [comment]
I think that it doesn't make a significant difference; in both cases, there's an extraneous tab stop that doesn't provide value to the user.

**Comment by annezazu** [comment]
Does this still need accessibility feedback? How can we move it forward? I'd like to add a `needs dev` label and add to the Polish board :)

**Comment by joedolson** [comment]
I think the accessibility feedback has been given. I'm not aware of any additional feedback required.

**Comment by alexstine** [comment]
Adjusted labels and moved into project. This should be set for dev.

**Comment by annezazu** [comment]
Fantastic, thank you.

**Comment by krokodok** [comment]
So, I'm going through this right now at WCEU Krakow three years later, trying to understand, what's the best way to improve this.

As explained in the Chrome documentation linked earlier (https://chromestatus.com/feature/5231964663578624), having the Tabstop for the scroll container seems to be desired browser behaviour and should not be changed by us. Additionaly, not all browsers seem to have adapted this behaviour. Strangely, I can only reproduce it in Firefox currently on MacOS, not even in Chrome.

So if we can't and shouldn't remove the tab stop, what can we do, to decrease screenreader & keyboard UX confusion? 

**Step 1** would be to have a label announcing what kind of container is receiving the keyboard focus. I don't know right now if this is was newly introduced during the last three years, but as of my testing today, the container does have a sensible label.

<img width="409" height="72" alt="Image" src="https://github.com/user-attachments/assets/b6497dc4-353d-4bb7-90f1-d539c7d56189" />

I can't reproduce the scrollbar in the floating inserter panel right now, but if this issue still persists in some configurations, this one would still be _missing_ a proper label.

<img width="598" height="176" alt="Image" src="https://github.com/user-attachments/assets/f1e5bb43-1004-41be-976f-65daf4c03f77" />

An `aria-labelledby` using the already present `h2` could be a solution.

**A second must-have** improvement would be to have a visible focus-state for the tabstop, since currently it's simply missing. It should look like this (mockup):

<img width="355" height="708" alt="Image" src="https://github.com/user-attachments/assets/e041fb4a-fe47-4019-9151-916d6d7c59e1" />

This would lessen user confusion without removing the seemingly necessary tabstop. Do you agree on the actionables?

- [ ] Find out, if scrollbar in floating inserter still exists
- [ ] Add accessible label to floating inserter
- [ ] Add focus-visible-states to scroll-containers with tabstops
